### ISSUE TITLE
False positives from one-component-per-file

### ISSUE THREAD

**Issue by markrian** [vuejs/eslint-plugin-vue#1411]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 7.17 (also tested with 6.8)
- **eslint-plugin-vue version:** 7.4.1
- **Node version:** 12.18.4
- **Operating System:** Ubuntu 20.04

**Please show your full configuration:**
```js
module.exports = {
  extends: ['plugin:vue/recommended'],
};
```

**What did you do?**	

```js
const foo = { mixin() {} };

Vue.component({
  mixins: [foo.mixin({})],
});
```

and equally:

```vue
<script>
const foo = { mixin() {} };

export default {
  mixins: [foo.mixin({})],
};
</script>
```

**What did you expect to happen?**
There should be no reported violations of one-component-per-file.

**What actually happened?**
```
yarn run v1.22.5
$ eslint test.js

/home/markrian/dev/eslint-plugin-vue-bug/test.js
  4:15  warning  There is more than one component in this file  vue/one-component-per-file
  5:22  warning  There is more than one component in this file  vue/one-component-per-file

✖ 2 problems (0 errors, 2 warnings)

Done in 0.39s.
```

**Repository to reproduce this issue**
https://gitlab.com/markrian/eslint-plugin-vue-one-component-per-file-bug

**Comment by markrian** [comment]
The really odd thing with this is that:

1. The called method _must_ be called `mixin`
2. The method _must_ be given an object literal as an argument

That is, none of these (separately) trigger the rule's warning:

```js
foo.bar({})
foo.extend({})
foo.mixin()
foo.mixin(new Object())
mixin({})
```

but this does:

```js
foo.mixin({})
```

**Comment by markrian** [comment]
After looking into this a bit further, I think this false positive ultimately comes from [this check](https://github.com/vuejs/eslint-plugin-vue/blob/f478d6554dd5aa11b21fd33be7b7ccc70d02c5bb/lib/utils/index.js#L1852-L1859):

```js
        // for Vue.js 3.x
        // app.component('xxx', {}) || app.mixin({})
        const isFullVueComponent =
          propName &&
          ['component', 'mixin'].includes(propName) &&
          isObjectArgument(node)

        return Boolean(isFullVueComponent)
```
